File: frontend/src/microservices/attraction/objects/CreateAttractionTicketSessionPlannerRequest.ts
export type CreateAttractionTicketSessionPlannerRequest = {
  managerId: string
  attractionId: string
  ticketTypeId: string
  sessionName: string
  useDate: string
  startsAt: string
  endsAt: string
  capacity: number
}

export const createAttractionTicketSessionPlannerRequestFromJson = (json: string): CreateAttractionTicketSessionPlannerRequest =>
  JSON.parse(json) as CreateAttractionTicketSessionPlannerRequest

export const createAttractionTicketSessionPlannerRequestToJson = (value: CreateAttractionTicketSessionPlannerRequest): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/attraction/objects/CreateAttractionTicketSessionPlannerRequest.scala
package com.typesafe.travel.attraction.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class CreateAttractionTicketSessionPlannerRequest(
    managerId: String,
    attractionId: String,
    ticketTypeId: String,
    sessionName: String,
    useDate: String,
    startsAt: String,
    endsAt: String,
    capacity: Int
)
object CreateAttractionTicketSessionPlannerRequest:
  given sourceEncoder: Encoder[CreateAttractionTicketSessionPlannerRequest] = deriveEncoder
  given sourceDecoder: Decoder[CreateAttractionTicketSessionPlannerRequest] = deriveDecoder
